opensearch/src/main/java/org/opensearch/sql/opensearch/request/OpenSearchRequestBuilder.java

```diff
@@ -237,4 +245,72 @@ private boolean isSortByDocOnly() {
     }
     return false;
   }
+
+  /**
+   * Push down nested to sourceBuilder.
+   * @param nestedArgs : Nested arguments to push down.
+   */
+  public void pushDownNested(List<Map<String, ReferenceExpression>> nestedArgs) {
```

Comment:
It seems like you reset the filter here, so filter set by `WHERE` clause is ignored.
```
opensearchsql> select *, nested(message.info), message.info from nested-type where myNum > 3;
fetched rows / total rows = 8/8
+-----------------------------------------+---------+-------------+-----------+------------------------+----------------+
| comment                                 | myNum   | someField   | message   | nested(message.info)   | message.info   |
|-----------------------------------------+---------+-------------+-----------+------------------------+----------------|
| {'data': 'ab', 'likes': 3}              | 1       | b           | null      | a                      | a              |
| {'data': 'aa', 'likes': 2}              | 2       | a           | null      | b                      | b              |
| {'data': 'aa', 'likes': 3}              | 3       | a           | null      | c                      | c              |
| {'data': 'ab', 'likes': 1}              | 4       | b           | null      | c                      | c              |
| {'data': 'ab', 'likes': 1}              | 4       | b           | null      | a                      | a              |
| {'data': ['aa', 'bb'], 'likes': 10}     | [3,4]   | a           | null      | zz                     | zz             |
| {'data': ['aa', 'bb'], 'likes': 10}     | [3,4]   | a           | null      | zz                     | zz             |
| {'data': ['asdf', 'sdfg'], 'likes': 56} | [1,2,4] | ert         | null      | rr                     | rr             |
+-----------------------------------------+---------+-------------+-----------+------------------------+----------------+
```

Reply:
Thanks, see [9ae39a114b48816f2dd0699f3fa431d7970042a8](https://github.com/opensearch-project/sql/pull/1490/commits/9ae39a114b48816f2dd0699f3fa431d7970042a8) for the fix.